pontoon/base/views.py

```diff
@@ -98,6 +98,30 @@ def translate(request, locale, slug, part):
     })
 
 
+# LOCALE-AGNOSTIC VIEWS
+
+
+def translate_locale_agnostic(request, slug, part):
+    """Locale Agnostic view."""
+    user = request.user
+    project = get_object_or_404(Project.objects.available(), slug=slug)
+
+    if user.is_authenticated():
+        locale = user.profile.custom_homepage
+
+        if locale and project.locales.filter(code=locale).exists():
+            return redirect('pontoon.translate', slug=slug, locale=locale, part=part)
+        else:
+            return redirect('pontoon.projects.project', slug=slug)
+    else:
+        locale = utils.get_project_locale_from_request(request, project.locales) or 'en-GB'
+
+        if not project.locales.filter(code=locale).exists():
+            locale = 'en-GB'
+
+        return redirect('pontoon.translate', slug=slug, locale=locale, part=part)
```

Comment:
Thanks for the update, that looks good now! But we should make two additional changes.

1. We should try the accept language locale also for authenticated users which don't have a custom homepage set or it's not enabled for the project.

2. There's also no point in hardcoding the redirect to the en-GB locale.

So the entire function should look like this:

```python
user = request.user
project = get_object_or_404(Project.objects.available(), slug=slug)

if user.is_authenticated():
    locale = user.profile.custom_homepage
    if locale and project.locales.filter(code=locale).exists():
        return redirect('pontoon.translate', slug=slug, locale=locale, part=part)

locale = utils.get_project_locale_from_request(request, project.locales)

if locale:
    return redirect('pontoon.translate', slug=slug, locale=locale, part=part)
else:
    return redirect('pontoon.projects.project', slug=slug)
```